vaadin-card-flow-parent/vaadin-card-flow/src/main/java/com/vaadin/flow/component/card/Card.java

```diff
@@ -32,6 +48,337 @@
+        if (contentRoot != null && contentRoot.getChildCount() == 0) {
+            getElement().removeChild(contentRoot);
+            contentRoot = null;
+        }
+    }
+
+    @Override
+    public void removeAll() {
+        if (contentRoot != null) {
+            contentRoot.removeAllChildren();
+            getElement().removeChild(contentRoot);
+            contentRoot = null;
+        }
+    }
+
+    @Override
+    public void addComponentAtIndex(int index, Component component) {
+        Objects.requireNonNull(component, "Component should not be null");
+        if (index < 0) {
+            throw new IllegalArgumentException(
+                    "Cannot add a component with a negative index");
+        }
+        if (contentRoot == null) {
+            initContentRoot();
+        }
+        contentRoot.insertChild(index, component.getElement());
+    }
+
+    /**
+     * Sets the ARIA role attribute on the card.
+     *
+     * @param role
+     *            the ARIA role, or {@code null} to clear
+     */
+    public void setAriaRole(String role) {
+        if (role == null) {
+            getElement().removeAttribute("role");
+        } else {
+            getElement().setAttribute("role", role);
+        }
+    }
+
+    /**
+     * Gets the ARIA role attribute of the card.
+     *
+     * @return an optional ARIA role of the card if no ARIA role has been set
+     */
+    public Optional<String> getAriaRole() {
+        return Optional.ofNullable(getElement().getAttribute("role"));
+    }
+
+    @Override
+    protected void onAttach(AttachEvent attachEvent) {
+        super.onAttach(attachEvent);
+        checkFeatureFlag();
+    }
+
+    /**
+     * Gets the feature flags for the current UI.
+     * <p>
+     * Not private in order to support mocking
+     *
+     * @return the current set of feature flags
+     */
+    FeatureFlags getFeatureFlags() {
+        return FeatureFlags
+                .get(UI.getCurrent().getSession().getService().getContext());
+    }
+
+    /**
+     * Only for test use.
+     */
+    void setFeatureFlagEnabled() {
+        featureFlagEnabled = true;
+    }
+
+    /**
+     * Checks whether the Card component feature flag is active. Succeeds if the
+     * flag is enabled, and throws otherwise.
+     *
+     * @throws ExperimentalFeatureException
+     *             when the {@link FeatureFlags#CARD_COMPONENT} feature is not
+     *             enabled
+     */
+    private void checkFeatureFlag() {
+        boolean enabled = featureFlagEnabled
+                || getFeatureFlags().isEnabled(FeatureFlags.CARD_COMPONENT);
+        if (!enabled) {
+            throw new ExperimentalFeatureException();
+        }
+    }
+
+    private void setSlotContent(String slotName, Component slotContent) {
+        if (slotContent == null) {
+            SlotUtils.clearSlot(this, slotName);
+        } else {
+            SlotUtils.setSlot(this, slotName, slotContent);
```

Comment:
`setSlot` already does `clearSlot` and accepts `null` as slot content, so it seems `setSlotContent` could be entirely replaced with `SlotUtils.setSlot(this, slotName, slotContent)`. The `getSlotContent` could then be replaced with direct `SlotUtils.getChildInSlot(this, slotName)`

Reply:
Done.